parse_xml.py

```diff
@@ -0,0 +1,174 @@
+                       #r = response.json
+                       #status = r["responseHeader"]["status"]
+                       #if status == 0:
+                       #    logging.info("-- Solr update for file: "+docid+" complete")
+                       #else:
+                       #    logging.info("-- Solr error for doc: "+docid+" error: "+', '.join("{!s}={!r}".format(k,v) for (k,v) in rdict.items()))
+    except IOError as e:
+        logging.error("I/O error({0}): {1}".format(e.errno,e.strerror))
+    except:
+        logging.error("Unexpected error:", sys.exc_info()[0])
+        raise
+
+#this function contains the code for parsing the xml file
+#and writing the results out to a json file
+def processXML(fname):
+    try:
+        with open(os.path.abspath(fname)) as fd:
+            fn = changeExt(fname,'json')
+            doc = xmltodict.parse(fd.read())
+            records = doc['main']['DATA_RECORD']
+            for x in records:
+                docid = x.get('DOCUMENT_IMAGE_ID',x.get('DOCUMENT_NM'))
+                txtfn = os.path.join(os.path.dirname(fn),'PDF_image',docid+'.txt')
+                if os.path.isfile(txtfn):
+                    x['LAST_MODIFIED_TS'] = format_date(x, 'LAST_MODIFIED_TS')
+                    x['PATENT_ISSUE_DT'] = format_date(x, 'PATENT_ISSUE_DT')
+                    x['DECISION_MAILED_DT'] = format_date(x, 'DECISION_MAILED_DT')
+                    x['PRE_GRANT_PUBLICATION_DT'] = format_date(x, 'PRE_GRANT_PUBLICATION_DT')
+                    x['APPLICANT_PUB_AUTHORIZATION_DT'] = format_date(x, 'APPLICANT_PUB_AUTHORIZATION_DT')
+                    x['appid'] = x.pop('BD_PATENT_APPLICATION_NO')
+                    x['doc_date'] = x.pop('DOCUMENT_CREATE_DT')
+                    with open(txtfn) as dr:
+                        text = dr.read()
+                        x['textdata'] = text
+                else:
+                    logging.info("TXT file: "+docid+".txt  does not exist. JSON file creation skipped.")
+                    return 
+
+            #transform output to json and save to file with same name
+            with open(fn,'w') as outfile:
+                json.dump(doc,outfile)
+                logging.info("-- Processing of XML file complete")
+                #process new JSON file
+    except IOError as e:
+        logging.error("I/O error({0}): {1}".format(e.errno,e.strerror))
+    except:
+        logging.error("Unexpected error:", sys.exc_info()[0])
+        raise
+
+#validate date
+def validDate(s):
+    try:
+        datetime.strptime(s, "%Y%m%d")
+        return s
+    except ValueError:
+        msg = "Not a valid date: '{0}'.".format(s)
+        raise argparse.ArgumentTypeError(msg)
+
+def processFile(filename):
+    fn = changeExt(filename,'json')
+    if os.path.isfile(os.path.abspath(fn)):
+        if (args.solr):
+            logging.info("-- XML file: "+filename+" already processed.  Skipping Solr process.")
```

Comment:
Ahh, I see why it looks backward. Because the arg is named in what looks like a positive way when it's actually a negative. The argument should be named `--skip-solr`, otherwise it is confusing.